web-demo/static/index.html

```diff
@@ -77,66 +78,185 @@
 
     <link rel="stylesheet" href="//unpkg.com/codemirror@5.65.2/lib/codemirror.css">
     <script src="//unpkg.com/codemirror@5.65.2/lib/codemirror.js"></script>
+    <script src="//unpkg.com/codemirror@5.65.2/mode/scheme/scheme.js"></script>
     <script src="https://unpkg.com/@hpcc-js/wasm@2.13.1/dist/graphviz.umd.js" type="javascript/worker"></script>
     <script type="module">
         import { compressUrlSafe, decompressUrlSafe } from './lzma-url.mjs';
         import * as d3 from 'https://cdn.skypack.dev/pin/d3@v7.8.5-eC7TKxlFLay7fmsv0gvu/dist=es2020,mode=imports,min/optimized/d3.js';
         import 'https://cdn.skypack.dev/-/d3-graphviz@v5.1.0-TcGnvMu4khUzCpL7Wr2k/dist=es2020,mode=imports,min/optimized/d3-graphviz.js';
 
         let editor, output, examples;
+        let nextButton, prevButton, slideNumber, docsButton, githubButton, shareButton, examplesDropDown;
         console.log("Main script is running");
         let worker = new Worker("worker.js");
+
+        var state = {
+            isSlideMode: false,
+            slideIndex: 0,
+            hasRun: true,
+            contentBySexp: [],
+        };
+
         worker.onmessage = function (message) {
             console.log("got message", message);
             output.innerHTML = message.data.text;
+            output.scrollTop = output.scrollHeight;
             const graphContainer = d3.select("#graph-inner")
             const width = graphContainer.node().clientWidth;
             const height = graphContainer.node().clientHeight;
             const graphviz = graphContainer.graphviz({
                 // Fit graph to that size, so that all is visible
                 fit: true,
-                // Set to be as big as container
-                width,
-                height,
+                width: width,
+                height: height,
                 // Don't animate transitions between shapes for performance
                 tweenPaths: false,
                 tweenShapes: false,
             })
                 .transition(d3.transition().duration(2000))
                 .dot(message.data.dot).render();
-            // If we have made a zoom selection, reset that before transitioning
```

Comment:
@saulshanabrook why did we reset zoom here? it seems to work better in general without it